File: doc/codes/python/examples/api/command/set-pulse-width-table-example.py
import asyncio

from autd3 import Client, ClientConfig
from autd3.commands import SetPulseWidthTable
from autd3.geometry import Autd3, Geometry
from autd3_link_nop import Nop


async def main() -> None:
    geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])
    client = await Client.open(geometry, Nop(), ClientConfig())

    table = SetPulseWidthTable.default_table()

    builder = client.datagram_builder()
    builder.push(SetPulseWidthTable(table=table))
    frames = builder.build()
    for frame in frames:
        await client.send_checked(frame)

    await client.close()


asyncio.run(main())

File: doc/codes/rust/examples/api/command/set-pulse-width-table-example.rs
use anyhow::Result;

use autd3_rs::commands::SetPulseWidthTable;
use autd3_rs::geometry::{Autd3, Geometry};
use autd3_rs::{Client, ClientConfig};
use autd3_rs_link_nop::Nop;

#[tokio::main(flavor = "multi_thread")]
async fn main() -> Result<()> {
    let geometry = Geometry::new(vec![Autd3::default()]);
    let client = Client::open(&geometry, Nop, ClientConfig::default()).await?;

    let table = SetPulseWidthTable::default_table();

    let mut builder = client.datagram_builder();
    builder.push(SetPulseWidthTable { table: &table });
    let frames = builder.build()?;
    for frame in &frames {
        client.send_checked(frame).await?;
    }

    client.close().await?;
    Ok(())
}
